src/WebUI/Workloads/DeploymentsTable.tsx

```diff
@@ -338,21 +341,33 @@ export class DeploymentsTable extends React.Component<IDeploymentsTablePropertie
     }
 
     private _onImageClick = (imageId: string): void => {
-        const imageService = KubeFactory.getImageService();
-        imageService && imageService.getImageDetails(imageId).then(imageDetails => {
-            if (imageDetails) {
-                const payload: ISelectionPayload = {
-                    item: imageDetails,
-                    itemUID: "",
-                    showSelectedItem: true,
-                    selectedItemType: SelectedItemKeys.ImageDetailsKey
-                };
-                this._selectionActionCreator.selectItem(payload);
-            }
-        });
+        const showImageDetails = (imageDetails: IImageDetails): void => {
+            const payload: ISelectionPayload = {
+                item: imageDetails,
+                itemUID: "",
+                showSelectedItem: true,
+                selectedItemType: SelectedItemKeys.ImageDetailsKey
+            };
+            this._selectionActionCreator.selectItem(payload);
+        }
+
+        let imageDetails: IImageDetails | undefined = this._imageDetailsStore.getImageDetails(imageId);
```

Comment:
code can be kept in util or in store and use in all 3 places

let imageDetails: IImageDetails | undefined = this._imageDetailsStore.getImageDetails(imageId);
        });	        if (imageDetails) {
            showImageDetails(imageDetails);
        }
        else {
            const imageService = KubeFactory.getImageService();
            imageService && imageService.getImageDetails(imageId).then(imageDetails => {
                if (imageDetails) {
                    ActionsCreatorManager.GetActionCreator<ImageDetailsActionsCreator>(ImageDetailsActionsCreator).setImageDetails(imageDetails);
                    showImageDetails(imageDetails);
                }
            });
        }